File: test/end-to-end/Graphs/Java/Data/UnweightedNode.java
package graphs.data;

import graphs.data.INode;
import java.util.ArrayList;

/**
 * An unweighted, undirected node in a graph.
 */
public class UnweightedNode<T> implements INode<UnweightedNode<T>> {
    public T data;
    private ArrayList<UnweightedNode<T>> neighborNodes;

    public UnweightedNode(T data) {
        this.data = data;
        this.neighborNodes = new ArrayList<UnweightedNode<T>>();
    }

    public void addNeighbor(UnweightedNode<T> node) {
        this.neighborNodes.add(node);
    }

    public ArrayList<UnweightedNode<T>> getNeighborsInOrder() {
        return this.neighborNodes;
    }
}

File: test/end-to-end/Graphs/Python/data/unweighted_node.py

"""
An unweighted, undirected node in a graph.
"""
class UnweightedNode:

    def __init__(self, data):
        self.data = data
        self.__neighbor_nodes = []

    def add_neighbor(self, node):
        self.__neighbor_nodes.append(node)

    def get_neighbors_in_order(self):
        return self.__neighbor_nodes
